### PR TITLE
promdump: Automatic batch duration backoff

### PR DESCRIPTION
**Pull request by ionthegeek**
This PR implements automatic batch duration backoff for promdump. If Prometheus finds that a query would return too many samples, it will throw an error that can be caught. With these changes in place, the error will be caught and the batch duration will be halved (down to a minimum of 1s) and the dump retried.

### CODE REVIEW

File: promdump/promdump.go [hunk 1]
```diff
@@ -43,6 +47,20 @@ func writeFile(values *[]*model.SampleStream, fileNum uint) error {
...
 }
 
+func cleanFiles() error {
+	files, err := filepath.Glob(fmt.Sprintf("%s.[0-9][0-9][0-9][0-9][0-9]", *out))
+	if err != nil {
+		return err
+	}
+	for _, file := range files {
+		err = os.Remove(file)
+		if err != nil {
```
Comment (eugeneckim) [inline promdump/promdump.go:59]: Perhaps we can print something about the removes.  Per our conversation, maybe not ALL removes, but a summary of how many, etc.

**Review by eugeneckim — APPROVED** [review]
Go go go (intended)
Notes from conversation:
- Consider printing information about the cleanup function for visibility.
- We had discussions about maximum number of files and backstopping it at potentially 5 digits.

**Review by na6vj — APPROVED** [review]

**Review by na6vj — APPROVED** [review]

File: promdump/promdump.go (hunk 1 above)
Comment (ionthegeek) [reply promdump/promdump.go:59]: Updated to address this feedback.

**Review by na6vj — COMMENTED** [review]
Any chance you could get rid of the outer "for allBatchesFetched == false {"  loop ?
The inner loop checks the condition "batch <= batches" anyway, making "allBatchesFetched" superfluous.

**Review by eugeneckim — APPROVED** [review]
Let's go!

**Comment by ionthegeek** [comment]
> Any chance you could get rid of the outer "for allBatchesFetched == false {" loop ? The inner loop checks the condition "batch <= batches" anyway, making "allBatchesFetched" superfluous.

Unfortunately this isn't possible because we have to reset both the current batch count and total batch count if the batch duration changes and these are calculated up front before the inner loop starts. The only straightforward way to redo the whole shebang is to wrap it in an outer loop with an exit condition (`allBatchesFetched`) and break out of the inner loop if we have to recalculate.

**Pull request merged**